src/devices/src/virtio/net/device.rs

```diff
@@ -80,6 +85,28 @@ fn init_vnet_hdr(buf: &mut [u8]) {
     }
 }
 
+fn generate_mac_address() -> Option<MacAddr> {
+    let mut random_mac: [u8; MAC_ADDR_LEN] = [0; MAC_ADDR_LEN];
+    let (left, right) = random_mac.split_at_mut(3);
+    left.copy_from_slice(&KVM_OUI);
+    let file_result = File::open(Path::new(RANDOM_GENERATOR));
+    let mut file = match file_result {
+        Ok(file) => file,
+        Err(e) => {
+            error!("Error opening file {}: {}", RANDOM_GENERATOR, e);
+            return MacAddr::from_bytes(&random_mac);
+        }
+    };
+    match file.read_exact(right) {
+        Ok(_) => right,
+        Err(e) => {
+            error!("Error reading from file {}: {}", RANDOM_GENERATOR, e);
+            return MacAddr::from_bytes(&random_mac);
+        }
+    };
+    MacAddr::from_bytes(&random_mac)
```

Comment:
Since we already know that random_mac has the proper length, we can use:

```
MacAddr::from_bytes_unchecked(&random_mac)
```
This would give you back a MacAddr instead of an Option<MacAddr>.